packages/amazon-bedrock/src/convert-to-bedrock-chat-messages.ts

```diff
@@ -0,0 +1,95 @@
+import { LanguageModelV1Prompt } from '@ai-sdk/provider';
+import { BedrockChatPrompt } from './bedrock-chat-prompt';
+import { ImageBlock, ImageFormat } from '@aws-sdk/client-bedrock-runtime';
+
+export function convertToBedrockChatMessages(
+  prompt: LanguageModelV1Prompt,
+): BedrockChatPrompt {
+  const messages: BedrockChatPrompt = [];
+
+  for (const { role, content } of prompt) {
+    switch (role) {
+      case 'user': {
+        messages.push({
+          role: 'user',
+          content: content.map(part => {
+            if (part.type === 'image' && part.image instanceof URL) {
+              throw new Error(
+                'Image URLs are not supported in Bedrock. Please convert the image to a base64 string.',
+              );
+            }
+
+            return part.type === 'text'
+              ? { text: part.text }
+              : {
+                  image: {
+                    format: part.mimeType?.split('/')?.[1] as ImageFormat,
+                    source: {
+                      // TODO: support image URL
+                      bytes: part.image as Uint8Array,
```

Comment:
e.g. the anthropic provider supports this via image download

Reply:
Added image download